src/result_set.rs

```diff
@@ -0,0 +1,160 @@
+use crate::{
+    ast::ParameterizedValue,
+    error::Error,
+    transaction::{ColumnNames, ResultRow},
+};
+use std::collections::HashMap;
+
+/// Encapsulates a set of results and their respective column names.
+pub struct ResultSet {
+    pub(crate) rows: Vec<ResultRow>,
+    pub(crate) name_to_index: HashMap<String, usize>,
+}
+
+impl ResultSet {
+    /// Creates a new instance, bound to the given column names and result rows.
+    pub fn new(names: &ColumnNames, values: Vec<ResultRow>) -> ResultSet {
+        ResultSet {
+            name_to_index: Self::build_name_map(names),
+            rows: values,
+        }
+    }
+
+    /// Creates a lookup map for column names.
+    pub(crate) fn build_name_map(names: &ColumnNames) -> HashMap<String, usize> {
```

Comment:
I'd say the `HashMap` should not be leaked out but kept as an internal implementation detail. Even though it's only public inside the crate.